generator/internal/connection_generator.cc

```diff
@@ -251,7 +253,24 @@ class $connection_class_name$ {
                   // clang-format off
     "\n  virtual future<Status>\n",
     "\n  virtual future<StatusOr<$longrunning_deduced_response_type$>>\n"},
-   {"  $method_name$($request_type$ const& request);\n"}
+   {"  $method_name$($request_type$ const& request);\n\n"},
+                 // clang-format on
+                 {IsResponseTypeEmpty,
+                  // clang-format off
+      "  virtual Status\n",
+      "  virtual StatusOr<$longrunning_operation_type$>\n"},
+                 // clang-format on
+                 // TODO(#14344): Remove experimental tag.
+                 {"  $method_name$(google::cloud::ExperimentalTag, "
+                  "google::cloud::NoAwaitTag,"
+                  " $request_type$ const& request);\n\n"},
+                 {IsResponseTypeEmpty,
+                  // clang-format off
+      "  virtual future<Status>\n",
+      "  virtual future<StatusOr<$longrunning_deduced_response_type$>>\n"},
+                 // TODO(#14344): Remove experimental tag.
+     {"  $method_name$(google::cloud::ExperimentalTag,"
```

Comment:
I think I prefer to call these things: `AwaitCreateFoo(...)` or `WaitForCreateFoo(...)`.

It stops being an overload set. I don't think that's bad.

```cc
auto operation = CreateFoo(request);
auto foo = CreateFoo(..., *operation);
```

The second line isn't creating a foo. That already happened. It is waiting on a foo. That is why I like the different name.

It seems super unlikely, but it could open us up to RPC name clashes. If that is the concern, feel free to leave it as is.